system/security/shared/caching.hpp

```diff
@@ -135,32 +135,46 @@ class CachedUser
 // main cache that stores all user-specific caches (defined by CResPermissionsCache above)
 //
 static CriticalSection PCCritSect;//guards instance factory
-static CPermissionsCache* instance = nullptr;//accessed via CPermissionsCache::queryInstance()
+typedef map<string, CPermissionsCache*> MapCache;
+static MapCache g_mapCache;
```

Comment:
Not a good move to put statics in the header file. Most of the code here should really be in the cpp file anyway...